src/Storages/StorageMemory.cpp

```diff
@@ -102,7 +106,9 @@ class MemoryBlockOutputStream : public IBlockOutputStream
         metadata_snapshot->check(block, true);
         {
             std::lock_guard lock(storage.mutex);
```

Comment:
Here the mutex also looks unneeded.

Reply:
I think we need it to resolve concurrent insert, mutation, drop...

Reply:
I'm not sure... Concurrent INSERT and mutation is Ok: mutation will mutate previously inserted data and concurrently inserted data will be left unmutated.

For DROP there is table-level lock: IStorage::lockExclusively.

Reply:
Let's write a test for race-condition (you can find similar .sh tests).

Reply:
We already have race-condition test `01454_storagememory_data_race_challenge.sh`.